### DISCUSSION TITLE
In a Select's custom tag render, how can we access properties other than just value and label?

### DISCUSSION THREAD

**Discussion by AsuraKev** [ant-design/ant-design#44062] [Q&A]
const tagRender = (props: CustomTagProps) => {
  const { label, value, closable, onClose } = props;
  const onPreventMouseDown = (event: React.MouseEvent<HTMLSpanElement>) => {
    event.preventDefault();
    event.stopPropagation();
  };
  return (
    <Tag
      color={value}
      onMouseDown={onPreventMouseDown}
      closable={closable}
      onClose={onClose}
      style={{ marginRight: 3 }}
    >
      {label}
    </Tag>
  );
};

const App: React.FC = () => (
  <Select
    mode="multiple"
    tagRender={tagRender}
    defaultValue={['gold', 'cyan']}
    style={{ width: '100%' }}
    options={options}
  />
);

**Comment by zhangzhuang15** [comment]
It really confuses me. I also have the same problem when I use Cascader Component. If we have many leaf nodes from different branches, and these leaf nodes have same label, there's no way to make sure what each node means if we follow current `tageRender`  declaration. If you want a workaround, you have to build hash-map to grab the parent option of selected option, it's annoying and stupid.
